### ISSUE TITLE
Make modes hideable on status-area

### ISSUE THREAD

**Issue by aartaka** [atlas-engineer/nyxt#2302]
**Is your feature request related to a problem? Please describe.**

After #2253, there appeared a bunch of modes that supercede web-mode. Even when having one-letter glyphs, those take up too much space. How about hiding some of those altogether if one configures it?

*Describe the solution you'd like**
Add a boolean slot for whether the mode should be displayed in the status area. 

**Describe alternatives you've considered**
A hack, that mostly does the same is setting `glyph` to `""`. But this is not perfect, as there's still some spacing/padding between mode glyphs,  so it looks bad.

CC @aadcg

**Comment by Ambrevar** [comment]
Agreed and actually we already have an issue related to this, see https://github.com/atlas-engineer/nyxt/issues/2218.

**Comment by Ambrevar** [comment]
I mean, we can fix both together.

**Comment by aadcg** [comment]
Agree!

(@aartaka supersede is the right spelling :p)

**Comment by aartaka** [comment]
> Agree!
> 
> (@aartaka supersede is the right spelling :p)

Ah damn, I always get it wrong :/

**Comment by aadcg** [comment]
The documentation slot says:

> A glyph used to represent this mode, if unset, it will be dynamically calculated as the first letters of the mode name.



This is not true today. Was it true before? Do we care about it? To my mind, it's not very important.

CC @jmercouris (see 99604753de265142f08b73ccddbec231a462e7fb)

**Comment by aadcg** [comment]
@aartaka, the slot already exists - `visible-in-status-p`. The question is which should be hidden by default. I'll do it my way, and then we'll discuss in the PR.

**Comment by Ambrevar** [comment]
Which idea and which docstring?

**Comment by aadcg** [comment]
> Which idea and which docstring?

This docstring:

> A glyph used to represent this mode, if unset, it will be dynamically calculated as the first letters of the mode name.

The idea is that when glyphs aren't set explicitly, then the appropriate mode prefix is computed.

**Comment by Ambrevar** [comment]
Oh sure.

**Comment by jmercouris** [comment]
Yes, we have foregone this idea.
